### PR TITLE
Allow AtmosModel run with DryModel moisture

### PR DESCRIPTION
**Pull request by lcw**
I am not sure if this is the fix we want but something along these lines is needed to run the RTB model with DryModel moisture.

### CODE REVIEW

**Comment by simonbyrne** [comment]
I think this relates to something @charleskawczynski and I were discussing yesterday: does it makes sense to specify different boundary conditions on different components, and if so, how should we do it?

File: src/Atmos/Model/moisture.jl [hunk 1]
```diff
@@ -103,3 +106,8 @@ function flux_diffusive!(m::EquilMoist, flux::Grad, state::Vars, diffusive::Vars
   flux.ρe += diffusive.moisture.ρd_h_tot
   flux.moisture.ρq_tot = diffusive.moisture.ρd_q_tot
 end
+
+function atmos_boundarycondition!(bc::NoFluxBC, m::EquilMoist, stateP::Vars, diffP::Vars, auxP::Vars, nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t)
+  diffP.moisture.ρd_q_tot = zero(eltype(diffP.moisture.ρd_q_tot))
+  diffP.moisture.ρd_h_tot = zero(eltype(diffP.moisture.ρd_h_tot))
```
Comment (simonbyrne) [inline src/Atmos/Model/moisture.jl]: @asraero should we set both diffusive terms to be zero?

**Comment by lcw** [comment]
Simon Byrne <notifications@github.com> writes:

> I think this relates to something @charleskawczynski and I were
> discussing yesterday: does it makes sense to specify different
> boundary conditions on different components, and if so, how should we
> do it?

Good question, as @mwarusz says "boundary conditions are hard".

I guess this leads to more overarching issues on the design of the
boundary conditions interface. Like, how do we want to specify Neumann,
Dirichlet, ...

**Comment by charleskawczynski** [comment]
I'm working on something now that can hopefully help with this. But I think I'll need help with loop unrolling

**Comment by simonbyrne** [comment]
I think this should get fixed by #413

**Comment by charleskawczynski** [comment]
Ah, yes, [`q_pt`](https://github.com/climate-machine/CLIMA/pull/413/files#diff-6f6f3c257f7970bd031f46109bd72cceL22) is not used, and it doesn't make sense to call `PhasePartition` for a dry model. So this should fix `internal_energy`/`thermo_state` calls for `DryModel`.

**Comment by charleskawczynski** [comment]
Can we close this @lcw ?

**Comment by lcw** [comment]
@charleskawczynski yep!

**Pull request closed without merging**